modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/ParameterModelsLoaderDelegate.java

```diff
@@ -44,39 +44,41 @@
         ParameterDeclarer parameter;
-        if (extensionParameter.isRequired()) {
-          parameter = groupDeclarer.withRequiredParameter(extensionParameter.getName());
+        if (parameterParser.isRequired()) {
+          parameter = groupDeclarer.withRequiredParameter(parameterParser.getName());
         } else {
-          parameter = groupDeclarer.withOptionalParameter(extensionParameter.getName())
-              .defaultingTo(extensionParameter.getDefaultValue());
+          parameter = groupDeclarer.withOptionalParameter(parameterParser.getName())
+              .defaultingTo(parameterParser.getDefaultValue());
         }
 
-        final MetadataType metadataType = extensionParameter.getType();
+        final MetadataType metadataType = parameterParser.getType();
         parameter.ofType(metadataType)
-            .describedAs(extensionParameter.getDescription())
-            .withAllowedStereotypes(extensionParameter.getAllowedStereotypes())
-            .withRole(extensionParameter.getRole())
-            .withExpressionSupport(extensionParameter.getExpressionSupport());
+            .describedAs(parameterParser.getDescription())
+            .withAllowedStereotypes(parameterParser.getAllowedStereotypes())
+            .withRole(parameterParser.getRole())
+            .withExpressionSupport(parameterParser.getExpressionSupport());
 
-        if (extensionParameter.isComponentId()) {
+        if (parameterParser.isComponentId()) {
           parameter.asComponentId();
         }
 
-        if (extensionParameter.isConfigOverride()) {
+        if (parameterParser.isConfigOverride()) {
           parameter.asConfigOverride();
         }
 
-        if (extensionParameter.isExcludedFromConnectivitySchema()) {
+        if (parameterParser.isExcludedFromConnectivitySchema()) {
           parameter.withModelProperty(new ExcludeFromConnectivitySchemaModelProperty());
         }
 
-        extensionParameter.getLayoutModel().ifPresent(parameter::withLayout);
-        extensionParameter.getDslConfiguration().ifPresent(parameter::withDsl);
-        extensionParameter.getDeprecationModel().ifPresent(parameter::withDeprecation);
-        extensionParameter.getDisplayModel().ifPresent(parameter::withDisplayModel);
-        extensionParameter.getAdditionalModelProperties().forEach(parameter::withModelProperty);
+        parameterParser.getLayoutModel().ifPresent(parameter::withLayout);
+        parameterParser.getDslConfiguration().ifPresent(parameter::withDsl);
+        parameterParser.getDeprecationModel().ifPresent(parameter::withDeprecation);
+        parameterParser.getDisplayModel().ifPresent(parameter::withDisplayModel);
+        parameterParser.getAdditionalModelProperties().forEach(parameter::withModelProperty);
+        addSemanticTerms(parameter.getDeclaration(), parameterParser);
+        parameter.getDeclaration().getSemanticTerms().addAll(parameterParser.getSemanticTerms());
```

Comment:
these two lines seems confusing, aren't they doing the same?

Reply:
good catch. Fixed